### ISSUE TITLE
Add Interval math with DateTime

### ISSUE THREAD

**Issue by schveiguy** [denizzzka/dpq2#180]
A postgres Interval type cannot be mapped to a Duration, because it has a "months" field.

But all time math on DateTime involves Duration. There should at least be a way to add an Interval to a DateTime.

**Comment by denizzzka** [comment]
Initially, I preferred not to add conversions or interop methods, which can suddenly lead to a decrease in accuracy

Now I think for some frequently used ypes we can add such conversion, with careful check that accuracy/range isn't lost (corresponding exception type already [implemented](
https://github.com/denizzzka/dpq2/blob/e9d1b33a61756fc29726fae8f40e0cb7137132f2/src/dpq2/value.d#L134
))

**Comment by schveiguy** [comment]
Yeah, I have this function in my code now. Not sure if std.datetime can deal with the overflow:
```d
@safe DateTime addInterval(DateTime date, dpq2.conv.time.Interval delta) {
    if(delta.months != 0) {
        date.add!"months"(delta.months);
    }
    // create a duration from the days/usecs
    import core.time : days, usecs;
    return date + (delta.days.days + delta.usecs.usecs);
}
```

I verified that posgres applies the months interval before the days/hours, so this should mimic that behavior.

**Comment by schveiguy** [comment]
Note that std.datetime also has an Interval type making this a bit awkward...

**Comment by denizzzka** [comment]
Looks good. Can you make PR?

If it's not urgent, then I can do it myself, but a little later

**Comment by schveiguy** [comment]
It's not urgent. I just wanted to note it as something that is needed. I can use our internal function. If it gets added, I will switch to using the official one.

**Comment by schveiguy** [comment]
If I find some time, I'll make a PR.
